torch/csrc/jit/passes/frozen_linear_transpose.cpp

```diff
@@ -0,0 +1,111 @@
+#include <torch/csrc/jit/ir/ir.h>
+#include <torch/csrc/jit/ir/ir_views.h>
+#include <torch/csrc/jit/jit_log.h>
+#include <torch/csrc/jit/passes/frozen_linear_transpose.h>
+#include <torch/csrc/jit/passes/utils/optimization_utils.h>
+#include <torch/csrc/jit/runtime/graph_executor.h>
+#include <iostream>
+
+namespace torch {
+namespace jit {
+namespace {
+
+using Tensor = at::Tensor;
+
+class TransposeFrozenLinear {
+ public:
+  TransposeFrozenLinear(std::shared_ptr<Graph> graph)
+      : graph_(std::move(graph)) {}
+
+  bool run() {
+    handleBlockAndSubblocks(graph_->block());
+    return graph_modified;
+  }
+
+  bool is_constant_linear_op(Node* node) {
+    if (node->kind() != aten::linear) {
+      return false;
+    }
+    auto weight = node->namedInput("weight");
```

Comment:
weight is never none, do you mean bias ? Can we handle the no-bias case as well here ?

Reply:
looks like bias is actually handled